includes/DataObjects/Request.php

```diff
@@ -256,6 +259,11 @@ public function hasComments()
 		if ($this->hasCommentsResolved) {
 			return $this->hasComments;
 		}
+        
+		if (isset(self::$commentsResults[$this->id])) {
+		    $this->hasCommentsResolved = true;
+		    return $this->hasComments = self::$commentsResults[$this->id];
+		}
```

Comment:
Each request has it's own instance of this class, but we do nothing as far as I'm aware to prevent there being multiple instances of this class representing the same request. I can't think of any point (not checked) where we would actually want multiple objects representing the same database record, which is why I'm suggesting caching the entire object, rather than just specific fields. That way there's both memory and cpu time saved.